### PR TITLE
docs: g2plot charts with live examples PTL-1521

### PR DESCRIPTION
**Pull request by SzymonZur**
ATENTION! YOU SHOULD NOW BRANCH FROM PREPROD WHEN YOU UPDATE 

  - Please check the [internal contributions guide](https://www.notion.so/genesisglobal/Contributing-new-documentation-75953fb245f246ff872789035451a0c4) for details
  
__________

Thank you for contributing to the documentation.

Do the changes you have made apply to both Current and Previous versions?

Have you done a trial build to check all new or changed links?

Is there anything else you would like us to know?

**This week's exciting advice from the style guide**

- Write your headings in sentence case:

  - A good example
    This is a correct heading.
  - A Bad Example
    This is not a correct heading.

### CODE REVIEW

File: docs/001_develop/03_client-capabilities/006_charts/002_g2plot-chart/01_chart-types.mdx [hunk 1]
```diff
@@ -0,0 +1,938 @@
...
+<Tabs defaultValue="declaration" values={[{ label: 'Declaration', value: 'declaration', }, { label: 'Configuration', value: 'config', }, { label: 'Sample data', value: 'data', }]}>
+
+<TabItem value="declaration">
+```html
+<rapid-g2plot-chart 
+  type="area"
+  :config=${chartConfiguration}
+  :data=${chartData}
+></rapid-g2plot-chart>
+```
```
Comment (matteematt) [inline docs/001_develop/03_client-capabilities/006_charts/002_g2plot-chart/01_chart-types.mdx:161]: Can I just double check with you that the bindings don't need to be functions, I thought they usually do in FAST? Same for other ones in this file
```suggestion
<rapid-g2plot-chart 
  type="area"
  :config=${() => chartConfiguration}
  :data=${() => chartData}
></rapid-g2plot-chart>
```
```

**Review by matteematt — COMMENTED** [review]
Really great work 🚀 I have left a few relatively small comments to look at, and then once addressed we can get that merged.

There is an additional change I would like though, but we can do it after this is merged. You do talk about most of the attributes, properties etc. required. But can you please have some explicit sections like [this layout example](https://github.com/genesiscommunitysuccess/docs/pull/1897/files#diff-9609b462206f9697259ef7e226c93cbc32762f04618bdab4e9e490d2a05cbcc7R54) following the same format going through each individual event, slot, property, and attribute please. Notice that we explicitly say when something is missing rather than just omitting it

File: docs/001_develop/03_client-capabilities/006_charts/002_g2plot-chart/index.mdx [hunk 2]
```diff
@@ -0,0 +1,212 @@
...
+    width: 700,
+    xField: "value",
+    yField: "groupBy",
+    seriesField: "series",
+    legend: {
+      position: "top-left",
+    },
+  };
+
+  const chartData = [
```
[the remark this answers is not in the document]
Comment (SzymonZur) [reply docs/001_develop/03_client-capabilities/006_charts/002_g2plot-chart/index.mdx:63]: Not really, the data can be almost anything depending on what kind of chart it is and what we want to show.

File: docs/001_develop/03_client-capabilities/006_charts/002_g2plot-chart/index.mdx [hunk 3]
```diff
@@ -0,0 +1,212 @@
...
+<TabItem value="genesis">
+
+Declaration:
+```html
+<rapid-g2plot-chart 
+  type="bar"
+  :config=${(x) => x.chartConfiguration}
+  :data=${(x) => x.chartData}
+></rapid-g2plot-chart>
+```
```
[the remark this answers is not in the document]
Comment (SzymonZur) [reply docs/001_develop/03_client-capabilities/006_charts/002_g2plot-chart/index.mdx]: [chore: addressed PR feedback PTL-1521](https://github.com/genesiscommunitysuccess/docs/pull/1890/commits/dde6b0e7282dea5ddf27534f308f15eada56b133)

File: docs/001_develop/03_client-capabilities/006_charts/002_g2plot-chart/index.mdx
[the remark this answers is not in the document]
Comment (SzymonZur) [reply docs/001_develop/03_client-capabilities/006_charts/002_g2plot-chart/index.mdx:1]: [chore: addressed PR feedback PTL-1521](https://github.com/genesiscommunitysuccess/docs/pull/1890/commits/dde6b0e7282dea5ddf27534f308f15eada56b133) - as discussed, changed to a non-clickable panel

**Pull request merged**